src/domain/rate-limit-middleware.js

```diff
@@ -0,0 +1,16 @@
+import httpErrors from 'http-errors';
+
+const { TooManyRequests } = httpErrors;
+
+function createRateLimitMiddleware({ maxRequests, expiresInMs, store }) {
+  return async function rateLimitMiddleware(req, _res, next) {
+    try {
+      const newCount = await store.incrementCount({ req, expiresInMs });
+      return newCount > maxRequests ? next(new TooManyRequests()) : next();
```

Comment:
I was about to point out that in the `handlePostUserLogin` method the condition is `>= FAILED_LOGIN_ATTEMPTS_LIMIT` and here the limit is exclusive (`>`) but then I realized this middleware is not used..right? 🤔

Reply:
Yeah, it's a little bit tricky, but this middleware is first updating the count for each request and then deciding if it should block or not. The `handlePostUserLogin` just serves the request and then updates the count afterwards (it has to because it only updates for *failed* login attempts, not for each request). So if we set the limit to `5` for example, in both situations the user has 5 shots before it blocks the next time.

Reply:
Ah yes, to actually answer your question, the middleware is not used in `handlePostUserLogin`

Reply:
What I meant is I cannot see where it is used at all.